src/components/Modal/Modal.tsx

```diff
@@ -1,130 +1,169 @@
     super(props);
-    this.state = {
-      active: false,
-    };
   }
 
-  handleCloseClick = (event: React.SyntheticEvent<HTMLElement> | KeyboardEvent): void => {
-    event.preventDefault();
-    this.setState({ active: false });
+  public id = getUniqueID();
+
+  componentDidMount() {
+    this.setAccessibilityAttributes();
   }
 
-  handleToggleClick = (e: React.SyntheticEvent<HTMLElement>): void => {
-    if (!this.state.active) {
-      this.setState({ active: true });
-    } else {
-      const target = e.target as HTMLInputElement;
-      const id = (target ? target.dataset.id : undefined);
-      if (typeof id !== 'undefined') {
-        const prefix = id.substr(0, id.indexOf('-'));
-        if (prefix === 'modal' && this.props.closeOnBackgroud) {
-          this.handleCloseClick(e);
-        }
-      }
+  componentDidUpdate() {
+    this.setAccessibilityAttributes();
+  }
+
+  componentWillReceiveProps(newProps: any) {
+    const { active } = this.props;
+    // Call the callback function if available
+    // onOpen: when drawer open
+    // onClose: when drawer close
+    if (!active && newProps.active && typeof newProps.onOpen !== 'undefined') {
+      newProps.onOpen();
+    } else if (active && !newProps.active && typeof newProps.onClose !== 'undefined') {
+      newProps.onClose();
     }
   }
 
-  render() {
+  getModalContainerClassName() {
+    const { theme, width } = this.props;
 
-    const {
-      activator,
-      children,
-      className,
-      close,
-      closeOnBackgroud,
-      closeOnEsc,
-      footer,
-      header,
-      id,
-      modalOverflow,
-      overlay = true,
-      width,
-      theme,
-    } = this.props;
-
-    const modalClassNames = classNames(
-      overlay && theme.overlay,
-      this.state.active && theme.open,
-      className
+    return classNames(
+      theme.dialog,
+      width === 'small' && theme.small,
+      width === 'medium' && theme.medium,
+      width === 'large' && theme.large
     );
+  }
 
-    const bodyClassNames = classNames(
-      theme.body,
-      modalOverflow && theme.autoHeight
+  getModalWrapperClassName() {
+    const { active, className, theme } = this.props;
+
+    return classNames(
+      theme.overlay,
+      active && theme.open,
+      className
     );
+  }
 
-    const bodyElement = document.body;
-    if (bodyElement !== null) {
-      bodyElement.className = this.state.active ? (theme.page) : '';
-    }
+  render() {
+    return null;
+  }
+
+  // Render close button if its set true
+  renderCloseButton() {
+    const { closeButton, theme } = this.props;
+
+    return closeButton ? (<div className={theme.close}><Button plain onClick={this.closeModal.bind(this)} icon="cancel" /></div>) : null;
+  }
+
+  // Import keylistener if modal needs to be closed on pressing escape key
+  renderKeyListener() {
+    return this.props.closeOnEsc ? (<KeypressListener keyCode={Keys.ESCAPE} handler={this.triggerCloseEvent.bind(this)} />) : null;
```

Comment:
@dheeraja00 Instead of bind, use arrow functions

Reply:
Check my above reply on similar comment